code/data_processor_block/data_processor.py

```diff
@@ -0,0 +1,60 @@
+#Jakub Podolak for Plastic Monkeys Cansat 2019
+
+import graphyte
+from time import sleep
+import ast
+import configparser
+import logging
+import sys
+
+config_file = sys.argv[1]
+config = configparser.ConfigParser()
+config.read(config_file)
+
+graphyte_address = config['SETTINGS']['graphyte_address']
+graphyte_prefix = config['SETTINGS']['graphyte_prefix']
+sleep_time = float(config['SETTINGS']['sleep_time'])
+
+graphyte.init(graphyte_address, prefix=graphyte_prefix)
+
+#open log file
+logging.basicConfig(filename="data_processor.log",level=logging.INFO)
+
+#function to read paths to input logs from config file
+def get_directories(config):
+    paths = config.items("PATHS")
+    directories = [second for first,second in paths]
+    return directories
+
+#extracts dict from string packet
+def trim_line(line):
+    dict_len = int(line[0:2])
+    return line[2:2+dict_len]
+
+#converts string-dict to dict
+def get_dict(line):
+    return ast.literal_eval(line)
+
+def send_to_graphite(data_dict):
+    for name,val in data_dict.items():
+        graphyte.send(name,val)
+
+def run():
+    directories = get_directories(config)
+
+    while True:
+        try:
+            for input_log in directories:
+                file = open(input_log, "r")
+                file.seek(0,2)
+                file.seek(file.tell()-41, 0) #Packet should have a size of 40 characters
```

Comment:
how do we know whether there were just one line added to the file? This approach might result in skipping lines, which we should avoid I believe?

Reply:
TBH I'm afraid, that any kind of going back in the log file is potentially dangerous. Imagine a situation when our base station receives packets more frequently than we can send it to graphite. In such a scenario, we are not displaying current values, but can actually get stuck with old data. In my opinion printing current data is more important than sending everything.

Reply:
If we store hard copy of received packets, I totally agree

Reply:
yup, all received packets are hard logging'ed